src/proto/graplinc/grapl/api/plugin_work_queue/v1beta1/plugin_work_queue.proto

```diff
@@ -39,33 +43,61 @@ message GetExecuteGeneratorRequest {}
 // A request to retrieve a new ExecutionJob for an Analyzer
 message GetExecuteAnalyzerRequest {}
 
+message NoAvailableJobs {}
+
 // A response containing the next ExecutionJob to execute against a Generator
 message GetExecuteGeneratorResponse {
   // The execution_job to be executed
-  ExecutionJob execution_job = 1;
+  oneof maybe_job {
```

Comment:
I'm not 100% opposed to it, but, ultimately, this feels like a reinvention of an optional field (yes, I know everything is optional in proto3 and they're a pain to work with). Is this the preferred pattern?

Reply:
It is indeed. Google provides the 'Empty' structure for this reason, but in this case I went with a custom one as it's more extendable (for example if we wanted to add a "try again after x seconds", etc).

protobuf sux